src/components/ToolTip/ToolTip.jsx

```diff
@@ -0,0 +1,197 @@
+import _ from 'lodash';
+import React from 'react';
+import ContextMenu from '../ContextMenu/ContextMenu';
+import * as reducers from './ToolTip.reducers';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createClass, findTypes } from '../../util/component-types';
+
+const cx = lucidClassNames.bind('&-ToolTip');
+
+const {
+	bool,
+	func,
+	node,
+	object,
+	oneOf,
+	string
+} = React.PropTypes;
+
+const {
+	Target,
+	FlyOut
+} = ContextMenu;
+
+
+/**
+ * {"categories": ["communication"], "madeFrom": ["ContextMenu"]}
+ *
+ * `ToolTip` is a utility component to create a transient message anchored
+ * to another component
+ */
+
+const ToolTip = createClass({
+	displayName: 'ToolTip',
+
+	reducers,
+
+	propTypes: {
+		/**
+		 * `children` should include exactly one ToolTip.Target and one ToolTip.FlyOut.
+		 */
+		children: node,
+		/**
+		 * Appended to the component-specific class names set on the root element.
+		 */
+		className: string,
+		/**
+		 * Passed through to the root element.
+		 */
+		style: object,
+		/**
+		 * direction of the FlyOut relative to Target. Defaults to `'down'`.
+		 */
+		direction: oneOf(['down', 'up', 'right', 'left']),
+		/**
+		 * alignment of the Flyout relative to Target in the cross axis from `direction` Defaults to `'start'`
+		 */
+		alignment: oneOf(['start', 'center', 'end']),
+		/**
+		 * Indicates whether the ToolTip will render or not. Defaults to `true`.
+		 */
+		isExpanded: bool,
```

Comment:
i just kept this since it maps to `ContextMenu`'s `isExpanded`, but should it be `isShown` here instead?